tests/conftest.py

```diff
@@ -232,3 +233,50 @@ def mock_fetch(monkeypatch, mock_importlib):
     monkeypatch.setattr(transaction, "_get_pypi_json", result._get_pypi_json)
     monkeypatch.setattr(transaction, "fetch_bytes", result._fetch_bytes)
     return result
+
+
+@pytest.fixture(scope="session")
+def docker_compose_file():
+    return str(Path(__file__).parent.parent / "devpi/docker-compose.yml")
+
+
+@pytest.fixture(scope="session")
+def devpi(docker_ip, docker_services):
```

Comment:
IMO we should be able to run the test suite without relying on docker: so if would be better if this was opt-in.